src/scramble_parser.rs

```diff
@@ -65,7 +65,41 @@ fn get_variant(mv: &str) -> MoveVariant {
     }
 }
 
-/// Simplifies the given moves by combining adjacent moves.
+/// Recursively merges adjacent moves of the same Move discriminant
+/// until no further simplification is possible.
 pub fn simplify_moves(moves: &[Move]) -> Vec<Move> {
-    vec![]
+    use std::mem::discriminant;
+    let mut result = vec![];
+    if moves.is_empty() {
+        return result;
+    }
+
+    // keep track of the current move and its amount of clockwise turns
+    type Movement = (Move, u8);
+    let mut movement: Movement = (moves[0], moves[0].get_variant() as u8);
```

Comment:
Instead of destructuring the Move, it may be better to just call the methods to extract the value when needed. This way, we don't need to refer back to our type definition to understand what ``movement.0`` and ``movement.1`` are. 

An alternative may be to use a helper function. E.g. ``getVariantAsU8(mv: Move) -> u8``.